### PR TITLE
mgr/PyModuleRegistry: log error if we can't find any modules to load

### PR DESCRIPTION
**Pull request by tserong**
Fixes: https://tracker.ceph.com/issues/39662
Signed-off-by: Tim Serong <tserong@suse.com>

- [x] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

File: src/mgr/PyModuleRegistry.cc [hunk 1]
```diff
@@ -67,6 +67,10 @@ void PyModuleRegistry::init()
   std::list<std::string> failed_modules;
 
   std::set<std::string> module_names = probe_modules();
+  if (module_names.empty()) {
+    // This should never happen, but will if mgr_module_path is set incorrectly
+    clog->error() << "No ceph-mgr modules found";
+  }
```
Comment (tchaikov) [inline src/mgr/PyModuleRegistry.cc]: i'd suggest move this hunk down after we load the python modules. i.e., to put it before the `if (!failed_modules.empty()) {` block. and might want to refactor `probe_modules()` to pass the path to it, and include the path in the log message, like
```c++
const std::string module_path = g_conf().get_val<std::string>("mgr_module_path");
for (const auto& module_name : probe_modules(module_path)) {
 // ...
}
if (modules.empty()) {
  clog->error() << "No ceph-mgr modules loaded in " << module_path;
}
if (!failed_modules.empty()) {
  // ...
}
```

File: src/mgr/PyModuleRegistry.cc [hunk 2]
```diff
@@ -259,6 +263,8 @@ std::set<std::string> PyModuleRegistry::probe_modules() const
 {
   std::string path = g_conf().get_val<std::string>("mgr_module_path");
 
+  dout(1) << "Looking for modules in '" << path << "'" << dendl;
```
Comment (tchaikov) [inline src/mgr/PyModuleRegistry.cc]: might want to drop this logging message.

File: src/mgr/PyModuleRegistry.cc (hunk 1 above)
Comment (tserong) [reply src/mgr/PyModuleRegistry.cc]: Yeah, that's a bit neater, thanks :-)

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2019-05-30_03:19:00-rados-wip-kefu-testing-2019-05-30-0110-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2019-05-30_03:19:00-rados-wip-kefu-testing-2019-05-30-0110-distro-basic-smithi/

**Pull request merged**